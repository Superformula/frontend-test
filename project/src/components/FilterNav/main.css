.filter-nav {
  padding: 25px 70px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;

  display: flex;
  justify-content: space-between;
}