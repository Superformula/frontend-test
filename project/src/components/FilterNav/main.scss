.filter-nav {
  padding: 15px 70px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;

  display: flex;
  justify-content: space-between;
}

.filter-nav .controls {
  display: flex;
  align-items: center;
}

.filter-nav .controls > * {
  margin-right: 20px;
}