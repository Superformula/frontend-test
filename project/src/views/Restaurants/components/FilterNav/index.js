import React, { useContext } from "react";
import styled from "styled-components";

import { AppContext } from "AppContext";
import FlatButton from "components/FlatButton";
import CheckBox from "components/FormControls/CheckBox";
import Select from "components/FormControls/Select";

const Wrapper = styled.div`
  padding: 15px 70px;
  border-top: 1px solid #eee;
  border-bottom: 1px solid #eee;
  display: flex;
  justify-content: space-between;
`;

const Controls = styled.div`
  display: flex;
  align-items: center;
  > * {
    margin-right: 20px;
  }
`;

const FilterNav = () => {
  const appContext = useContext(AppContext);
  const {
    setPrice,
    setOpenNow,
    setCategory,
    clearAll,
    openNow,
    price,
    category
  } = appContext;

  return (
    <Wrapper>
      <Controls>
        <span>Filter By:</span>
        <CheckBox label="Open Now" onChange={setOpenNow} checked={openNow} />
        <Select
          onChange={setPrice}
          placeholder="Price"
          options={["All", "$", "$$", "$$$", "$$$$"]}
          value={price}
        />
        <Select
          onChange={setCategory}
          placeholder="Categories"
          options={[
            "Italian",
            "Seafood",
            "Steakhouses",
            "Japanese",
            "American",
            "Mexican",
            "Thai"
          ]}
          value={category}
        />
      </Controls>
      <FlatButton onClick={clearAll}>Clear All</FlatButton>
    </Wrapper>
  );
};

export default FilterNav;
