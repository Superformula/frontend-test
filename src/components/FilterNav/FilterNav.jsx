import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  GridItem,
  Container,
  Checkbox,
  SelectDropDown,
  Button,
} from '..';

const FilterHeading = styled.span`
  color: ${props => props.theme.colors.black60};
`;

class FilterNav extends PureComponent {
  handleFilter = (filter, value) => {
    const { toggleFilter } = this.props;
    toggleFilter(filter, value);
  };

  render() {
    const {
      openNow,
      price,
      category,
      toggleCategory,
    } = this.props;
    return (
      <GridItem
        width={[1]}
        borderTop="1px solid"
        borderBottom="1px solid"
        borderColor="black10"
        py={4}
      >
        <Container>
          <GridItem
            width={[1 / 2, 3 / 4, 3 / 4]}
            display="inline-flex"
          >
            <GridItem
              display="inline-block"
              mr={4}
            >
              <FilterHeading>Filter By:</FilterHeading>
            </GridItem>
            <GridItem
              display="inline-block"
              mr={4}
            >
              <Checkbox
                checked={openNow}
                onChange={() => this.handleFilter('openNow', !openNow)}
                name="openNow"
                id="openNow"
                label="Open Now"
              />
            </GridItem>
            <GridItem
              width={97}
              display="inline-block"
              mr={4}
            >
              <SelectDropDown
                heading="Price"
                items={[
                  { title: 'All', value: '' },
                  { title: '$', value: '$' },
                  { title: '$$', value: '$$' },
                  { title: '$$$', value: '$$$' },
                  { title: '$$$$', value: '$$$$' },
                ]}
                selectedItem={price}
                onSelect={item => this.handleFilter('price', item.value)}
              />
            </GridItem>
            <GridItem
              width={193}
              display="inline-block"
            >
              <SelectDropDown
                heading="Categories"
                items={[
                  { title: 'All', value: '' },
                  { title: 'Italian', value: 'italian' },
                  { title: 'Seafood', value: 'seafood' },
                  { title: 'Steakhouses', value: 'steakhouses' },
                  { title: 'Japanese', value: 'japanese' },
                  { title: 'American', value: 'american' },
                  { title: 'Mexican', value: 'mexican' },
                  { title: 'Thai', value: 'thai' },
                ]}
                selectedItem={category}
                onSelect={item => toggleCategory(item.value)}
              />
            </GridItem>
          </GridItem>
          <GridItem
            width={[1 / 2, 1 / 4, 1 / 4]}
            display="inline-flex"
            justifyContent="flex-end"
          >
            <Button variant="secondary" small>Clear Filters</Button>
          </GridItem>
        </Container>
      </GridItem>
    );
  }
}

FilterNav.propTypes = {
  toggleFilter: PropTypes.func.isRequired,
  toggleCategory: PropTypes.func.isRequired,
  openNow: PropTypes.bool.isRequired,
  price: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
};

export default FilterNav;
