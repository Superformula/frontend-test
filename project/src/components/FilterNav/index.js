import React from "react";

import FlatButton from "components/FlatButton";
import CheckBox from "components/FormControls/CheckBox"

require("./main.scss");

const FilterNav = ({ setPrice, setOpenNow, setCategory }) => {
  return (
    <div className="filter-nav">
      <div className="controls">
        <span>Filter By:</span>

        <CheckBox 
          label="Open Now"
          onChange={setOpenNow}
        /> 
        
        <select defaultValue="" onChange={e => setPrice(e.target.value)}>
          <option value="" disabled hidden>
            Price
          </option>
          <option value={"All"}>All</option>
          <option value={"$"}>$</option>
          <option value={"$$"}>$$</option>
          <option value={"$$$"}>$$$</option>
          <option value={"$$$$"}>$$$$</option>
        </select>

        <select defaultValue="" onChange={e => setCategory(e.target.value)}>
          <option value="" disabled hidden>
            Categories
          </option>
          <option value={"Italian"}>Italian</option>
          <option value={"Seafood"}>Seafood</option>
          <option value={"Steakhouses"}>Steakhouses</option>
          <option value={"Japanese"}>Japanese</option>
          <option value={"American"}>American</option>
          <option value={"Mexican"}>Mexican</option>
          <option value={"Thai"}>Thai</option>
        </select>

      </div>

      <FlatButton>Clear All</FlatButton>
    </div>
  );
};

export default FilterNav;
