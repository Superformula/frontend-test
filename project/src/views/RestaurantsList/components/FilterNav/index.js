import React, { useContext } from "react";

import { AppContext } from "AppContext";
import FlatButton from "components/FlatButton";
import CheckBox from "components/FormControls/CheckBox";
import Select from "components/FormControls/Select";

require("./main.scss");

const FilterNav = () => {
  const appContext = useContext(AppContext);

  const {
    setPrice,
    setOpenNow,
    setCategory,
    clearAll,
    location,
    openNow,
    price,
    category
  } = appContext;

  return (
    <div className="filter-nav">
      <div className="controls">
        <span>Filter By:</span>
        <CheckBox 
          label="Open Now" 
          onChange={setOpenNow} 
          checked={openNow}
        />
        <Select
          onChange={setPrice}
          placeholder="Price"
          options={["All", "$", "$$", "$$$", "$$$$"]}
          value={price}
        />
        <Select
          onChange={setCategory}
          placeholder="Categories"
          options={[
            "Italian",
            "Seafood",
            "Steakhouses",
            "Japanese",
            "American",
            "Mexican",
            "Thai"
          ]}
          value={category}
        />
      </div>

      <FlatButton onClick={clearAll}>Clear All</FlatButton>
    </div>
  );
};

export default FilterNav;
