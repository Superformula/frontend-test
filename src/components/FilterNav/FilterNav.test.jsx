import React from 'react';
import { mountWithTheme } from '../../helpers/tests/withTheme';
import FilterNav from './FilterNav';

const defaultProps = {
  toggleFilter: jest.fn(),
  toggleCategory: jest.fn(),
  openNow: true,
  price: '$$',
  category: 'italian',
};

describe('FilterNav Component', () => {
  let component;
  it('renders itself primary button correctly', () => {
    component = mountWithTheme(<FilterNav {...defaultProps}>Primary</FilterNav>).find('FilterNav');
    expect(component).toMatchSnapshot();
  });
});
