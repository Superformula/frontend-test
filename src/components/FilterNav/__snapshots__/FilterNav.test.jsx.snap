// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterNav Component renders itself primary button correctly 1`] = `
<FilterNav
  category="italian"
  openNow={true}
  price="$$"
  toggleCategory={[MockFunction]}
  toggleFilter={[MockFunction]}
>
  <styled.div
    borderBottom="1px solid"
    borderColor="black10"
    borderTop="1px solid"
    py={4}
    width={
      Array [
        1,
      ]
    }
  >
    <div
      className="sc-gqjmRU hwYUaN"
      width={
        Array [
          1,
        ]
      }
    >
      <styled.div>
        <div
          className="sc-iwsKbI oPSfI"
        >
          <styled.div
            display="inline-flex"
            width={
              Array [
                0.5,
                0.75,
                0.75,
              ]
            }
          >
            <div
              className="sc-gqjmRU kiHGsi"
              display="inline-flex"
              width={
                Array [
                  0.5,
                  0.75,
                  0.75,
                ]
              }
            >
              <styled.div
                display="inline-block"
                mr={4}
              >
                <div
                  className="sc-gqjmRU cGHpIG"
                  display="inline-block"
                >
                  <styled.span>
                    <span
                      className="sc-ckVGcZ bGWCTB"
                    >
                      Filter By:
                    </span>
                  </styled.span>
                </div>
              </styled.div>
              <styled.div
                display="inline-block"
                mr={4}
              >
                <div
                  className="sc-gqjmRU cGHpIG"
                  display="inline-block"
                >
                  <Checkbox
                    checked={true}
                    id="openNow"
                    isDisabled={false}
                    label="Open Now"
                    name="openNow"
                    onChange={[Function]}
                  >
                    <styled.div>
                      <div
                        className="sc-jTzLTM efNPAV"
                      >
                        <styled.label
                          borderBottom="1px solid"
                          borderColor="black21"
                          htmlFor="openNow"
                          pb={2}
                        >
                          <label
                            className="sc-VigVT iomSMd"
                            htmlFor="openNow"
                          >
                            <styled.input
                              checked={true}
                              id="openNow"
                              isDisabled={false}
                              name="openNow"
                              onChange={[Function]}
                            >
                              <input
                                checked={true}
                                className="sc-fjdhpX hcUlkV"
                                id="openNow"
                                name="openNow"
                                onChange={[Function]}
                                type="checkbox"
                              />
                            </styled.input>
                            <styled.div
                              border="1px solid"
                              borderColor="black21"
                              checked={true}
                            >
                              <div
                                checked={true}
                                className="sc-jzJRlG jXBXWB"
                              >
                                <styled.svg
                                  checked={true}
                                  viewBox="0 0 24 24"
                                >
                                  <svg
                                    checked={true}
                                    className="sc-cSHVUG blGJjt"
                                    viewBox="0 0 24 24"
                                  >
                                    <circle
                                      cx="12"
                                      cy="12"
                                      r="10"
                                    />
                                  </svg>
                                </styled.svg>
                              </div>
                            </styled.div>
                            <span>
                              Open Now
                            </span>
                          </label>
                        </styled.label>
                      </div>
                    </styled.div>
                  </Checkbox>
                </div>
              </styled.div>
              <styled.div
                display="inline-block"
                mr={4}
                width={97}
              >
                <div
                  className="sc-gqjmRU kvVSH"
                  display="inline-block"
                  width={97}
                >
                  <OnClickOutside(SelectDropDown)
                    eventTypes={
                      Array [
                        "mousedown",
                        "touchstart",
                      ]
                    }
                    excludeScrollbar={false}
                    heading="Price"
                    items={
                      Array [
                        Object {
                          "title": "All",
                          "value": "",
                        },
                        Object {
                          "title": "$",
                          "value": "$",
                        },
                        Object {
                          "title": "$$",
                          "value": "$$",
                        },
                        Object {
                          "title": "$$$",
                          "value": "$$$",
                        },
                        Object {
                          "title": "$$$$",
                          "value": "$$$$",
                        },
                      ]
                    }
                    onSelect={[Function]}
                    outsideClickIgnoreClass="ignore-react-onclickoutside"
                    preventDefault={false}
                    selectedItem="$$"
                    stopPropagation={false}
                  >
                    <SelectDropDown
                      disableOnClickOutside={[Function]}
                      enableOnClickOutside={[Function]}
                      eventTypes={
                        Array [
                          "mousedown",
                          "touchstart",
                        ]
                      }
                      heading="Price"
                      items={
                        Array [
                          Object {
                            "title": "All",
                            "value": "",
                          },
                          Object {
                            "title": "$",
                            "value": "$",
                          },
                          Object {
                            "title": "$$",
                            "value": "$$",
                          },
                          Object {
                            "title": "$$$",
                            "value": "$$$",
                          },
                          Object {
                            "title": "$$$$",
                            "value": "$$$$",
                          },
                        ]
                      }
                      onSelect={[Function]}
                      outsideClickIgnoreClass="ignore-react-onclickoutside"
                      preventDefault={false}
                      selectedItem="$$"
                      stopPropagation={false}
                    >
                      <styled.div
                        position="relative"
                      >
                        <div
                          className="sc-gqjmRU bAkkMC"
                        >
                          <styled.div
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <div
                              className="sc-dxgOiQ dWBkrD"
                              onClick={[Function]}
                              onKeyPress={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              Price
                              <styled.span>
                                <span
                                  className="sc-kGXeez kYdYIQ"
                                >
                                  <SvgChevronDown>
                                    <svg
                                      data-file-name="SvgChevronDown"
                                    />
                                  </SvgChevronDown>
                                </span>
                              </styled.span>
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                    </SelectDropDown>
                  </OnClickOutside(SelectDropDown)>
                </div>
              </styled.div>
              <styled.div
                display="inline-block"
                width={193}
              >
                <div
                  className="sc-gqjmRU cejnGY"
                  display="inline-block"
                  width={193}
                >
                  <OnClickOutside(SelectDropDown)
                    eventTypes={
                      Array [
                        "mousedown",
                        "touchstart",
                      ]
                    }
                    excludeScrollbar={false}
                    heading="Categories"
                    items={
                      Array [
                        Object {
                          "title": "All",
                          "value": "",
                        },
                        Object {
                          "title": "Italian",
                          "value": "italian",
                        },
                        Object {
                          "title": "Seafood",
                          "value": "seafood",
                        },
                        Object {
                          "title": "Steakhouses",
                          "value": "steakhouses",
                        },
                        Object {
                          "title": "Japanese",
                          "value": "japanese",
                        },
                        Object {
                          "title": "American",
                          "value": "american",
                        },
                        Object {
                          "title": "Mexican",
                          "value": "mexican",
                        },
                        Object {
                          "title": "Thai",
                          "value": "thai",
                        },
                      ]
                    }
                    onSelect={[Function]}
                    outsideClickIgnoreClass="ignore-react-onclickoutside"
                    preventDefault={false}
                    selectedItem="italian"
                    stopPropagation={false}
                  >
                    <SelectDropDown
                      disableOnClickOutside={[Function]}
                      enableOnClickOutside={[Function]}
                      eventTypes={
                        Array [
                          "mousedown",
                          "touchstart",
                        ]
                      }
                      heading="Categories"
                      items={
                        Array [
                          Object {
                            "title": "All",
                            "value": "",
                          },
                          Object {
                            "title": "Italian",
                            "value": "italian",
                          },
                          Object {
                            "title": "Seafood",
                            "value": "seafood",
                          },
                          Object {
                            "title": "Steakhouses",
                            "value": "steakhouses",
                          },
                          Object {
                            "title": "Japanese",
                            "value": "japanese",
                          },
                          Object {
                            "title": "American",
                            "value": "american",
                          },
                          Object {
                            "title": "Mexican",
                            "value": "mexican",
                          },
                          Object {
                            "title": "Thai",
                            "value": "thai",
                          },
                        ]
                      }
                      onSelect={[Function]}
                      outsideClickIgnoreClass="ignore-react-onclickoutside"
                      preventDefault={false}
                      selectedItem="italian"
                      stopPropagation={false}
                    >
                      <styled.div
                        position="relative"
                      >
                        <div
                          className="sc-gqjmRU bAkkMC"
                        >
                          <styled.div
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <div
                              className="sc-dxgOiQ dWBkrD"
                              onClick={[Function]}
                              onKeyPress={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              Categories
                              <styled.span>
                                <span
                                  className="sc-kGXeez kYdYIQ"
                                >
                                  <SvgChevronDown>
                                    <svg
                                      data-file-name="SvgChevronDown"
                                    />
                                  </SvgChevronDown>
                                </span>
                              </styled.span>
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                    </SelectDropDown>
                  </OnClickOutside(SelectDropDown)>
                </div>
              </styled.div>
            </div>
          </styled.div>
          <styled.div
            display="inline-flex"
            justifyContent="flex-end"
            width={
              Array [
                0.5,
                0.25,
                0.25,
              ]
            }
          >
            <div
              className="sc-gqjmRU iREoxV"
              display="inline-flex"
              width={
                Array [
                  0.5,
                  0.25,
                  0.25,
                ]
              }
            >
              <styled.button
                block={false}
                secondary={false}
                small={true}
                variant="secondary"
              >
                <button
                  className="sc-bxivhb ceBfMQ"
                >
                  Clear Filters
                </button>
              </styled.button>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</FilterNav>
`;
